framework/packages/abstract-std/src/native_addrs.rs

```diff
@@ -1,59 +1,64 @@
 
 pub use cw_blob::CHECKSUM as BLOB_CHECKSUM;
 
-// TODO: fill bytes with Public address of creator
-// Default local-juno used right now(for testing)
-const TEST_ABSTRACT_CREATOR: [u8; 33] = [
-    2, 146, 187, 207, 156, 96, 230, 188, 163, 167, 152, 64, 234, 101, 130, 38, 50, 89, 139, 233,
-    56, 192, 110, 242, 251, 222, 103, 198, 68, 80, 201, 159, 3,
-];
-
 // Salts for deployments
 pub const ANS_HOST_SALT: &[u8] = b"ans";
 pub const REGISTRY_SALT: &[u8] = b"reg";
 pub const MODULE_FACTORY_SALT: &[u8] = b"mf";
 
-pub fn ans_address(hrp: &str, api: &dyn Api) -> AbstractResult<CanonicalAddr> {
-    contract_canon_address(hrp, ANS_HOST_SALT, api)
+pub fn ans_address(deps: Deps, abstract_code_id: u64) -> AbstractResult<CanonicalAddr> {
+    contract_canon_address(deps, abstract_code_id, ANS_HOST_SALT)
 }
 
-pub fn registry_address(hrp: &str, api: &dyn Api) -> AbstractResult<CanonicalAddr> {
-    contract_canon_address(hrp, REGISTRY_SALT, api)
+pub fn registry_address(deps: Deps, abstract_code_id: u64) -> AbstractResult<CanonicalAddr> {
+    contract_canon_address(deps, abstract_code_id, REGISTRY_SALT)
 }
 
-pub fn module_factory_address(hrp: &str, api: &dyn Api) -> AbstractResult<CanonicalAddr> {
-    contract_canon_address(hrp, MODULE_FACTORY_SALT, api)
+pub fn module_factory_address(deps: Deps, abstract_code_id: u64) -> AbstractResult<CanonicalAddr> {
+    contract_canon_address(deps, abstract_code_id, MODULE_FACTORY_SALT)
 }
 
-pub fn derive_addr_from_pub_key(hrp: &str, pub_key: &[u8]) -> AbstractResult<String> {
-    let hrp: Hrp = Hrp::parse(hrp)?;
-
-    let hash = Sha256::digest(pub_key);
-    let rip_hash = Ripemd160::digest(hash);
+/// Address of the abstract admin
+pub fn creator_address(
+    querier: &cosmwasm_std::QuerierWrapper,
+    abstract_code_id: u64,
+) -> cosmwasm_std::StdResult<Addr> {
+    let code_info = querier.query_wasm_code_info(abstract_code_id)?;
+    Ok(code_info.creator)
+}
 
-    let addr = bech32::encode::<Bech32>(hrp, &rip_hash)?;
+pub fn creator_canon_address(deps: Deps, abstract_code_id: u64) -> StdResult<CanonicalAddr> {
+    // TODO: should we color it with unsafe?
+    // It's safe unless multithreaded or multi-chain
+    // https://google.github.io/comprehensive-rust/unsafe-rust/mutable-static.html
 
-    Ok(addr)
-}
+    // Uncomment to use
 
-/// Address of the abstract admin
-pub fn creator_address(hrp: &str) -> AbstractResult<String> {
-    derive_addr_from_pub_key(hrp, &TEST_ABSTRACT_CREATOR)
+    // static mut CREATOR_CANON_ADDRESS: Option<CanonicalAddr> = None;
+    // unsafe {
+    //     if CREATOR_CANON_ADDRESS.is_none() {
+    //         CREATOR_CANON_ADDRESS = Some(
+    creator_address(&deps.querier, abstract_code_id).and_then(|creator_addr| {
+        deps.api
+            .addr_canonicalize(creator_addr.as_str())
+            .map(Into::into)
+    })
+    //             ?,
+    //         )
+    //     }
+    //     Ok(CREATOR_CANON_ADDRESS.clone().unwrap())
+    // }
 }
```

Comment:
Commented part breaks on multichain, would also be unsafe in multi-threaded(not applicable to chains). What is ideal outcome for this?

Reply:
Ideal outcome would be : 
- Working inside tests with multi-chain
- Able to cache this (and more) queries during contract execution to save gas

Reply:
We could work with once-cell and hashmaps, but not sure this is the best idea

Reply:
OnceLock makes it safe in multi thread. Don't think we supposed to use hashmap(unless feature flagged?) for storing creator in static.